src/base/emu-i386/simx86/cpu-emu.c

```diff
@@ -1244,6 +1267,16 @@ int e_in_compiled_code(void)
     return InCompiledCode;
 }
 
+int EMU_V86(void)
+{
+    return (config.cpu_vm == CPUVM_EMU || (CEmuStat & CeS_INSTREMU));
+}
+
+int EMU_DPMI(void)
+{
+    return (config.cpu_vm_dpmi == CPUVM_EMU || (CEmuStat & CeS_INSTREMU));
```

Comment:
There are a few places, mostly via IS_EMU(). I could export IS_EMU() instead, and eliminate EMU_DPMI() though?

As for `EMU_DPMI()` directly:
`e_dpmi_b0x(op,scp);` could also be called unconditionally with a check inside;
the check in dnative `__init_handler` can simply be removed as that can't be true anymore.

`EMU_V86()` isn't called anywhere directly.

Reply:
Yes, I'd rather not export this EMU_DPMI()
as we have one already in emu.h, and unless
really needed, its probably better to leave
as is. And if we can dynamically switch the
back-ends, then it seems more logical to
remove even the existing IS_EMU() and
related macros, assume that emu is always
there, and use the inner checks rather than
the outer checks.
But not insisting, that just what comes to
the mind first.

Reply:
I replaced the check for async signals by if (!in_dpmi_emu && !in_vm86_emu) which works independently of the config backend.

Reply:
Thanks, looks good.